src/data-workspace/data-value-mutations/mutation-key-factory.js

```diff
@@ -0,0 +1,18 @@
+const rootKey = ['dataValues']
+
+const createSubKey = (subScope) => (params) => {
+    const theState = {
+        ...subScope,
+    }
+    if (params) {
+        theState.params = params
+    }
+    return [...rootKey, { ...theState }]
+}
+
+export const mutationKeys = {
+    all: createSubKey(),
+    update: createSubKey({ method: 'update' }),
+    file: createSubKey({ method: 'update', entity: 'file' }),
+    delete: createSubKey({ method: 'delete' }),
+}
```

Comment:
Maybe I'm dumb lol, but took me a while to figure out what these keys would look like given the higher-order function -- suggestion to make it easy to understand for other dummies:

```suggestion
/**
 * Example keys:
 * mutationKeys.all() ==> ['dataValues']
 * mutationKeys.update(dvParams)
 *     ==> ['dataValues', { method: 'update', params: dvParams }]
 * mutationKeys.file(dvParams)
 *     ==> ['dataValues', { method: 'update', entity: 'file', params: dvParams }]
 * etc.
 */
export const mutationKeys = {
    all: createSubKey(),
    update: createSubKey({ method: 'update' }),
    file: createSubKey({ method: 'update', entity: 'file' }),
    delete: createSubKey({ method: 'delete' }),
}
```

Reply:
Agreed, was looking to comment this a bit more. A bonus is that you can use them "without" the params. Eg. `mutationKeys.update()` targets all `update`-mutations.

Reply:
What's wrong with simply writing out these keys?

```js
const sharedKey = 'dataValues'

/**
 * MutationKeys for dataValues.
 * These are all functions that can be called with the dataValueParams
 * to narrow down the mutation
 */
export const mutationKeys = {
    all: (params) => [sharedKey, { params }],
    update: (params) => [sharedKey, { method: 'update', params }],
    file: (params) => [sharedKey, { method: 'update', entity: 'file', params }],
    delete: (params) => [sharedKey, { method: 'delete', params }],
}
```

That seems to be a lot easier to read (at least to me) and reduces the complexity of this module.

Reply:
I don't want `undefined` params, since when I tested it - it didn't seem to work properly with key-matching. 

Your second solution doesn't really work, since you're spreading the entire `params` object, not creating a nested `param` property.

Reply:
I got confused by the name "key" and didn't think of it referring to the "mutation key" but as the key of a vanilla js object.. Everything clear to me now!